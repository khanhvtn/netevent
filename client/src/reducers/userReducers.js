import {
  USER_CHECK,
  USER_LOGIN,
  USER_LOADING,
  USER_CHECKING,
  USER_CONFIRM,
  USER_IS_CONFIRM,
  FETCH_ALL_USERS,
  FETCH_CURRENT_USER,
  USER_LOGOUT,
  USER_PICK_ROLE,
  USER_CREATE_SUCCESS,
  USER_UPDATE_SUCCESS,
  USER_DELETE_SUCCESS,
  GET_ALL_USERS
} from '../constants';

const initialState = {
  isUserChecking: false,
  isLoading: false,
  isConfirm: false,
  pickedRoleNum: null,
  users: [],
  user: null,
  totalPages: null,
  isCreated: false,
  isUpdated: false,
  isDeleted: false
};

export default function userReducers(state = initialState, action) {
  switch (action.type) {
    // AUTH
    case USER_LOGIN:
      return { ...state, user: action.payload };
    case USER_LOGOUT:
      return initialState;
    case USER_LOADING:
      return { ...state, isLoading: action.payload };
    case USER_PICK_ROLE:
      return { ...state, pickedRoleNum: action.payload };

    // CONFIRMATION
    case USER_CONFIRM:
      return { ...state, user: action.payload };
    case USER_IS_CONFIRM:
      return { ...state, isConfirm: action.payload };
    case USER_CHECK:
      return { ...state, user: action.payload };
    case USER_CHECKING:
      return { ...state, isUserChecking: action.payload };

    // CRUD
    case FETCH_ALL_USERS:
      return {
        ...state,
        users: action.payload.data.data,
        totalPages: action.payload.data.totalPages
      };
    case GET_ALL_USERS:
      return {
        ...state,
        users: action.payload.data.data
      };
    case FETCH_CURRENT_USER:
      return {
        ...state,
        user: action.payload
      };

    // SNACKBAR
    case USER_CREATE_SUCCESS:
      return { ...state, isCreated: action.payload };
    case USER_UPDATE_SUCCESS:
      return { ...state, isUpdated: action.payload };
    case USER_DELETE_SUCCESS:
      return { ...state, isDeleted: action.payload };
    default:
      return state;
  }
}
