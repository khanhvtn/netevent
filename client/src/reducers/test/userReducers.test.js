import {
    USER_CHECK,
    USER_LOGIN,
    USER_LOADING,
    USER_CHECKING,
    USER_CONFIRM,
    USER_IS_CONFIRM,
    FETCH_ALL_USERS,
    FETCH_CURRENT_USER,
    USER_LOGOUT,
    USER_PICK_ROLE,
    USER_CREATE_SUCCESS,
    USER_UPDATE_SUCCESS,
    USER_DELETE_SUCCESS,
    GET_ALL_USERS,
} from '../../constants';
import userReducers from '../userReducers'
import expect from 'expect'


describe('USER REDUCER', () => {
    const initialState = {
        isUserChecking: false,
        isLoading: false,
        isConfirm: false,
        pickedRoleNum: null,
        users: [],
        user: null,
        totalPages: null,
        isCreated: false,
        isUpdated: false,
        isDeleted: false,
    };
    it('should return default state', () => {
        const action = userReducers(undefined, {})
        expect(action).toMatchSnapshot(initialState)
    });
    it('should return USER_LOGIN', () => {
        const action = userReducers(initialState, {
            type: USER_LOGIN
        })
        expect(action).toMatchSnapshot({
            isUserChecking: false,
            isLoading: false,
            isConfirm: false,
            pickedRoleNum: null,
            users: [],
            user: action.payload,
            totalPages: null,
            isCreated: false,
            isUpdated: false,
            isDeleted: false,
        })
    });
    it('should return USER_LOGOUT', () => {
        const action = userReducers(initialState, {
            type: USER_LOGOUT
        })
        expect(action).toMatchSnapshot(initialState)
    });
    it('should return USER_LOADING', () => {
        const action = userReducers(initialState, {
            type: USER_LOADING
        })
        expect(action).toMatchSnapshot({
            isUserChecking: false,
            isLoading: action.payload,
            isConfirm: false,
            pickedRoleNum: null,
            users: [],
            user: null,
            totalPages: null,
            isCreated: false,
            isUpdated: false,
            isDeleted: false,
        })
    });
    it('should return USER_PICK_ROLE', () => {
        const action = userReducers(initialState, {
            type: USER_PICK_ROLE
        })
        expect(action).toMatchSnapshot({
            isUserChecking: false,
            isLoading: false,
            isConfirm: false,
            pickedRoleNum: action.payload,
            users: [],
            user: null,
            totalPages: null,
            isCreated: false,
            isUpdated: false,
            isDeleted: false,
        })
    });
    it('should return USER_CONFIRM', () => {
        const action = userReducers(initialState, {
            type: USER_CONFIRM
        })
        expect(action).toMatchSnapshot({
            isUserChecking: false,
            isLoading: false,
            isConfirm: false,
            pickedRoleNum: null,
            users: [],
            user: action.payload,
            totalPages: null,
            isCreated: false,
            isUpdated: false,
            isDeleted: false,
        })
    });
    it('should return USER_IS_CONFIRM', () => {
        const action = userReducers(initialState, {
            type: USER_IS_CONFIRM
        })
        expect(action).toMatchSnapshot({
            isUserChecking: false,
            isLoading: false,
            isConfirm: action.payload,
            pickedRoleNum: null,
            users: [],
            user: null,
            totalPages: null,
            isCreated: false,
            isUpdated: false,
            isDeleted: false,
        })
    });
    it('should return USER_CHECK', () => {
        const action = userReducers(initialState, {
            type: USER_CHECK
        })
        expect(action).toMatchSnapshot({
            isUserChecking: false,
            isLoading: false,
            isConfirm: false,
            pickedRoleNum: null,
            users: [],
            user: action.payload,
            totalPages: null,
            isCreated: false,
            isUpdated: false,
            isDeleted: false,
        })
    });
    it('should return USER_CHECKING', () => {
        const action = userReducers(initialState, {
            type: USER_CHECKING
        })
        expect(action).toMatchSnapshot({
            isUserChecking: action.payload,
            isLoading: false,
            isConfirm: false,
            pickedRoleNum: null,
            users: [],
            user: null,
            totalPages: null,
            isCreated: false,
            isUpdated: false,
            isDeleted: false,
        })
    });
    it('should return FETCH_CURRENT_USER', () => {
        const action = userReducers(initialState, {
            type: FETCH_CURRENT_USER
        })
        expect(action).toMatchSnapshot({
            isUserChecking: false,
            isLoading: false,
            isConfirm: false,
            pickedRoleNum: null,
            users: [],
            user: action.payload,
            totalPages: null,
            isCreated: false,
            isUpdated: false,
            isDeleted: false,
        })
    });
    it('should return USER_CREATE_SUCCESS', () => {
        const action = userReducers(initialState, {
            type: USER_CREATE_SUCCESS
        })
        expect(action).toMatchSnapshot({
            isUserChecking: false,
            isLoading: false,
            isConfirm: false,
            pickedRoleNum: null,
            users: [],
            user: null,
            totalPages: null,
            isCreated: action.payload,
            isUpdated: false,
            isDeleted: false,
        })
    });
    it('should return USER_UPDATE_SUCCESS', () => {
        const action = userReducers(initialState, {
            type: USER_UPDATE_SUCCESS
        })
        expect(action).toMatchSnapshot({
            isUserChecking: false,
            isLoading: false,
            isConfirm: false,
            pickedRoleNum: null,
            users: [],
            user: null,
            totalPages: null,
            isCreated: false,
            isUpdated: action.payload,
            isDeleted: false,
        })
    });
    it('should return USER_DELETE_SUCCESS', () => {
        const action = userReducers(initialState, {
            type: USER_DELETE_SUCCESS
        })
        expect(action).toMatchSnapshot({
            isUserChecking: false,
            isLoading: false,
            isConfirm: false,
            pickedRoleNum: null,
            users: [],
            user: null,
            totalPages: null,
            isCreated: false,
            isUpdated: false,
            isDeleted: action.payload,
        })
    });
})