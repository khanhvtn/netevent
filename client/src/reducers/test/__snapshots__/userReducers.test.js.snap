// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`USER REDUCER should return FETCH_CURRENT_USER 1`] = `
Object {
  "isConfirm": false,
  "isCreated": false,
  "isDeleted": false,
  "isLoading": false,
  "isUpdated": false,
  "isUserChecking": false,
  "pickedRoleNum": null,
  "totalPages": null,
  "user": undefined,
  "users": Array [],
}
`;

exports[`USER REDUCER should return USER_CHECK 1`] = `
Object {
  "isConfirm": false,
  "isCreated": false,
  "isDeleted": false,
  "isLoading": false,
  "isUpdated": false,
  "isUserChecking": false,
  "pickedRoleNum": null,
  "totalPages": null,
  "user": undefined,
  "users": Array [],
}
`;

exports[`USER REDUCER should return USER_CHECKING 1`] = `
Object {
  "isConfirm": false,
  "isCreated": false,
  "isDeleted": false,
  "isLoading": false,
  "isUpdated": false,
  "isUserChecking": undefined,
  "pickedRoleNum": null,
  "totalPages": null,
  "user": null,
  "users": Array [],
}
`;

exports[`USER REDUCER should return USER_CONFIRM 1`] = `
Object {
  "isConfirm": false,
  "isCreated": false,
  "isDeleted": false,
  "isLoading": false,
  "isUpdated": false,
  "isUserChecking": false,
  "pickedRoleNum": null,
  "totalPages": null,
  "user": undefined,
  "users": Array [],
}
`;

exports[`USER REDUCER should return USER_CREATE_SUCCESS 1`] = `
Object {
  "isConfirm": false,
  "isCreated": undefined,
  "isDeleted": false,
  "isLoading": false,
  "isUpdated": false,
  "isUserChecking": false,
  "pickedRoleNum": null,
  "totalPages": null,
  "user": null,
  "users": Array [],
}
`;

exports[`USER REDUCER should return USER_DELETE_SUCCESS 1`] = `
Object {
  "isConfirm": false,
  "isCreated": false,
  "isDeleted": undefined,
  "isLoading": false,
  "isUpdated": false,
  "isUserChecking": false,
  "pickedRoleNum": null,
  "totalPages": null,
  "user": null,
  "users": Array [],
}
`;

exports[`USER REDUCER should return USER_IS_CONFIRM 1`] = `
Object {
  "isConfirm": undefined,
  "isCreated": false,
  "isDeleted": false,
  "isLoading": false,
  "isUpdated": false,
  "isUserChecking": false,
  "pickedRoleNum": null,
  "totalPages": null,
  "user": null,
  "users": Array [],
}
`;

exports[`USER REDUCER should return USER_LOADING 1`] = `
Object {
  "isConfirm": false,
  "isCreated": false,
  "isDeleted": false,
  "isLoading": undefined,
  "isUpdated": false,
  "isUserChecking": false,
  "pickedRoleNum": null,
  "totalPages": null,
  "user": null,
  "users": Array [],
}
`;

exports[`USER REDUCER should return USER_LOGIN 1`] = `
Object {
  "isConfirm": false,
  "isCreated": false,
  "isDeleted": false,
  "isLoading": false,
  "isUpdated": false,
  "isUserChecking": false,
  "pickedRoleNum": null,
  "totalPages": null,
  "user": undefined,
  "users": Array [],
}
`;

exports[`USER REDUCER should return USER_LOGOUT 1`] = `
Object {
  "isConfirm": false,
  "isCreated": false,
  "isDeleted": false,
  "isLoading": false,
  "isUpdated": false,
  "isUserChecking": false,
  "pickedRoleNum": null,
  "totalPages": null,
  "user": null,
  "users": Array [],
}
`;

exports[`USER REDUCER should return USER_PICK_ROLE 1`] = `
Object {
  "isConfirm": false,
  "isCreated": false,
  "isDeleted": false,
  "isLoading": false,
  "isUpdated": false,
  "isUserChecking": false,
  "pickedRoleNum": undefined,
  "totalPages": null,
  "user": null,
  "users": Array [],
}
`;

exports[`USER REDUCER should return USER_UPDATE_SUCCESS 1`] = `
Object {
  "isConfirm": false,
  "isCreated": false,
  "isDeleted": false,
  "isLoading": false,
  "isUpdated": undefined,
  "isUserChecking": false,
  "pickedRoleNum": null,
  "totalPages": null,
  "user": null,
  "users": Array [],
}
`;

exports[`USER REDUCER should return default state 1`] = `
Object {
  "isConfirm": false,
  "isCreated": false,
  "isDeleted": false,
  "isLoading": false,
  "isUpdated": false,
  "isUserChecking": false,
  "pickedRoleNum": null,
  "totalPages": null,
  "user": null,
  "users": Array [],
}
`;
